impress-org/givewp
# Feature: add Constant Contact add-on to the form migration process

Resolves: [GIVE-176]
## Description

This PR includes a form migration step for the new Constant Contact block migrating from a v2 form to a v3 form. I also removed the commented section of supported addons for Constant Contact so that it will display on our list when upgrading.

 Cases when the block should be added.
- While  Constant Contact  is globally enabled on all forms, if the form is not explicitly disabled from the per form setting migrating should add the block.
- While  Constant Contact  is not globally enabled if it is explicitly enabled on the per form setting than migrating should add the block.

## Testing Instructions with updated visuals

https://github.com/impress-org/givewp/assets/75056371/f6ba3298-8a3f-409d-943d-53f4f030d586

https://github.com/impress-org/givewp/assets/75056371/c4c31306-fd11-4e6b-8a94-0e41e4b74854



- Use [New  Constant Contact  Block 
](https://github.com/impress-org/give-constant-contact/pull/42)
- Set your global settings
- add a v2 form -> then migrate the form without editing the form settings.
- Verify the v3 form has the correct settings set from the global options.
- Create another v2 form -> update the per-form settings on the right -> then migrate the form
- Verify the v3 form has the correct settings set from the form options.

## Pre-review ChecklistT




[GIVE-136]: https://stellarwp.atlassian.net/browse/GIVE-136?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-64]: https://stellarwp.atlassian.net/browse/GIVE-64?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[GIVE-176]: https://stellarwp.atlassian.net/browse/GIVE-176?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
